angularSTEP/src/app/current-profile-page/current-profile-page.component.ts

```diff
@@ -62,11 +46,66 @@ export class CurrentProfilePageComponent implements OnInit {
       .ref.withConverter(new Converter().userConverter)
       .get();
     if (postUser !== null && postUser.data() !== undefined) {
+      this.route.queryParams.subscribe(params => {
+        const tab = params['tab'];
+        switch (tab) {
+          case 'wishlist': {
+            this.selected = 1;
+            break;
+          }
+          case 'shoppinglist': {
+            this.selected = 2;
+            break;
+          }
+          default: {
+            this.selected = 0;
+          }
+        }
+      });
       this.zone.run(() => {
         this.user = postUser.data()!;
       })
     } else {
       this.router.navigate(['/login']);
     }
   }
+
+  editValue(form: string) {
+    switch (form) {
+      case 'displayName': {
+        this.displayNameForm = new FormControl(this.user!.displayName);
+        break;
+      }
+    }
+  }
+
+  cancelForm(form: string) {
+    switch (form) {
+      case 'displayName': {
+        this.displayNameForm = null;
+        break;
+      }
+    }
+  }
+
+  onSubmit() {
+    this.fAuth.currentUser.then(user => {
+      if (
+        user &&
+        this.displayNameForm &&
+        this.displayNameForm.valid &&
+        this.displayNameForm.value !== this.user!.aboutme
```

Comment:
Why compare to aboutme? I'm guessing you meant to compare to the current value of the display name.

Reply:
Oh yes, thanks!